runqslower.skel.rs-DELETEME

```diff
@@ -0,0 +1,167 @@
+    inner: &'a mut libbpf_rs::OpenObject,
+}
+
+impl<'a> OpenRunqslowerProgs<'a> {
+    pub fn handle__sched_wakeup(&mut self) -> &mut libbpf_rs::OpenProgram {
+        self.inner.prog_unwrap("handle__sched_wakeup")
+    }
+
+    pub fn handle__sched_wakeup_new(&mut self) -> &mut libbpf_rs::OpenProgram {
+        self.inner.prog_unwrap("handle__sched_wakeup_new")
+    }
+
+    pub fn handle__sched_switch(&mut self) -> &mut libbpf_rs::OpenProgram {
+        self.inner.prog_unwrap("handle__sched_switch")
+    }
+}
+
+pub struct OpenRunqslowerSkel {
+    pub obj: libbpf_rs::OpenObject,
+}
+
+impl OpenRunqslowerSkel {
+    pub fn load(&mut self) -> libbpf_rs::Result<RunqslowerSkel> {
+        Ok(RunqslowerSkel {
+            obj: self.obj.load()?,
+            links: None,
+        })
+    }
+
+    pub fn progs(&mut self) -> OpenRunqslowerProgs {
+        OpenRunqslowerProgs {
+            inner: &mut self.obj,
+        }
+    }
+
+    pub fn maps(&mut self) -> OpenRunqslowerMaps {
+        OpenRunqslowerMaps {
+            inner: &mut self.obj,
+        }
+    }
+}
+
+pub struct RunqslowerMaps<'a> {
+    inner: &'a mut libbpf_rs::Object,
+}
+
+impl<'a> RunqslowerMaps<'a> {
+    pub fn start(&mut self) -> &mut libbpf_rs::Map {
+        self.inner.map_unwrap("start")
+    }
+
+    pub fn min_us(&mut self) -> &mut libbpf_rs::Map {
+        self.inner.map_unwrap("min_us")
+    }
+
+    pub fn events(&mut self) -> &mut libbpf_rs::Map {
+        self.inner.map_unwrap("events")
+    }
+}
+
+pub struct RunqslowerProgs<'a> {
+    inner: &'a mut libbpf_rs::Object,
+}
+
+impl<'a> RunqslowerProgs<'a> {
+    pub fn handle__sched_wakeup(&mut self) -> &mut libbpf_rs::Program {
+        self.inner.prog_unwrap("handle__sched_wakeup")
+    }
+
+    pub fn handle__sched_wakeup_new(&mut self) -> &mut libbpf_rs::Program {
+        self.inner.prog_unwrap("handle__sched_wakeup_new")
+    }
+
+    pub fn handle__sched_switch(&mut self) -> &mut libbpf_rs::Program {
+        self.inner.prog_unwrap("handle__sched_switch")
+    }
+}
+
+pub struct RunqslowerLinks {
+    pub handle__sched_wakeup: libbpf_rs::Link,
+    pub handle__sched_wakeup_new: libbpf_rs::Link,
+    pub handle__sched_switch: libbpf_rs::Link,
+}
+
+pub struct RunqslowerSkel {
+    pub obj: libbpf_rs::Object,
+    pub links: Option<RunqslowerLinks>,
+}
+
+impl RunqslowerSkel {
+    pub fn progs(&mut self) -> RunqslowerProgs {
+        RunqslowerProgs {
+            inner: &mut self.obj,
+        }
+    }
+
+    pub fn maps(&mut self) -> RunqslowerMaps {
+        RunqslowerMaps {
+            inner: &mut self.obj,
+        }
+    }
+
+    pub fn attach(&mut self) -> libbpf_rs::Result<()> {
+        self.links = Some(RunqslowerLinks {
+            handle__sched_wakeup: self.progs().handle__sched_wakeup().attach()?,
+            handle__sched_wakeup_new: self.progs().handle__sched_wakeup_new().attach()?,
+            handle__sched_switch: self.progs().handle__sched_switch().attach()?,
```

Comment:
not all programs can be auto-attached, will this handle that? or it will return error? C skeleton skips attachment of program types that don't support auto-attachement (e.g., cgroup-based programs, they need cgroup FD, which can't be provided declaratively, so user-space has to provide extra parameters).

Reply:
I think libbpf returns `-ESRCH` if the program can't be autoattached